### DISCUSSION TITLE
Establishing an IKEv2/IPSec PSK connection from Xubuntu StrongSwan client to Draytek Vigor 2927ax router

### DISCUSSION THREAD

**Discussion by KlausStoertebeker** [strongswan/strongswan#1534] [Q&A]
Hello to the forum!
After searching for help in the internet for at least a week without success (also in this forum), contacting Draytek support (with the very motivating answer "_We do not have experience with linux and can't help you_") and frustrating trial-and-error without success, I try to get help here - and hopefully a solution.

My problem is as follows:

I'm running Xubuntu 22.04 (64 bit) and would like to establish an IKEv2/IPSec PSK connection from the Xubuntu strongSwan client to my Draytek Vigor 2927ax router. Therefore I installed the following strongSwan-packages:

`sudo apt-get -y install strongswan network-manager-strongswan strongswan-extra-plugins strongswan-pki strongswan-swanctl`)

I hoped to only use the gui of the Xubuntu strongSwan client to configure the IKEv2/IPSec PSK connection (see the following image - I haven't done ANY editing of configuration files via text editor in the file system). My Draytek router is available over the internet via the address "my-routers-domain.de" which uses DynDNS.

![Linux-NetworkManager-strongSwan-Configuration-Window](https://user-images.githubusercontent.com/13214286/218273159-fec1eb98-5d73-4750-a6c5-cb3efaef4939.png)

But I don't know which of the data I can setup in my Draytek Vigor 2927ax router I have to use. So in the following pictures you can see which data are available:

VPN-Services enabling:

![Draytek-Vigor2927ax-VPN-Serverdienste](https://user-images.githubusercontent.com/13214286/218273155-f0f36fd9-2b86-44da-bec2-8fce6bb12251.png)

General IKE/IPSec configuration:

![Draytek-Vigor2927ax-IPSec-Configuration](https://user-images.githubusercontent.com/13214286/218273150-e2a11255-2364-43f2-9b25-5769b978556e.png)

IPSec Peer Identity profile setup:

![Draytek-Vigor2927ax-IPSec-Peer-Identity](https://user-images.githubusercontent.com/13214286/218273152-2f2529fa-f06f-4f7b-9804-8dff11680160.png)

External dial-in user profile setup:

![Draytek-Vigor2927ax-External-User-Configuration](https://user-images.githubusercontent.com/13214286/218273139-335612f2-8b8f-4b71-b107-ea367d865092.png)

In addition, because I don't know if I'll need it, I created a server certificate signed by a CAcert certificate which I imported into the local certificate storage as well as I imported the CAcert root certificate (there is also an intermediate CAcert root certificate but I don't know if and in which storage I have to import it).
I t would be nice if someone could tell me which of the above data I have to use in the NetworkManager strongSwan configuration dialog to be able to connect to my Draytek router.
Thanks for help an greetings from Germany

"_KlausStoertebeker_"

**Comment by tobiasbrunner** [comment]
If you use PSK authentication, you don't need any certificates (neither a CA, nor a server certificate) as that's the whole point of using a shared secret.

So on the client you only need to configure the address of the router (hostname or IP address), optionally the server identity (otherwise the configured address will be used, probably corresponds to "Lokale ID" on the router), the client identity (username field on the client, probably corresponds to the "Peer-ID" field on the router) and, of course, the PSK. Note that the latter has to be at least 20 characters long, otherwise, the NM plugin will reject it.

Also be aware that we generally recommend using IKEv2 with EAP authentication and certificate authentication for the server in remote access scenarios (see [FAQ](https://docs.strongswan.org/docs/5.9/support/faq.html#_pre_shared_key_authentication)), which the Draytek router does seem to support. But as long as the PSK is strong and you are the only user, using a PSK should be fine.
